### ISSUE TITLE
BaseRequest.url passes the host and port to `URL.build` when the port is included in the host header (non-default port) -- fixed in 3.10.7

### ISSUE THREAD

**Issue by bdraco** [fafhrd91/aiohttp#9307]
request.host can be `127.0.0.1:8123`

https://github.com/aio-libs/aiohttp/blob/0b8be7f6c8db0d59afe02109df9be874d0d637e6/aiohttp/web_request.py#L447

yarl 1.13.0+ caught this because host is now validated via https://github.com/aio-libs/yarl/pull/954

Pinning yarl to < 1.13.0 will also prevent the issue since the validation doesn't happen until that version.

**Comment by bdraco** [comment]
Fix for Home Assistant is https://github.com/home-assistant/core/pull/126882

**Comment by bdraco** [comment]
This is messy because downstream now expects the port in a lot of places to get the host 
```
tests/components/http/test_forwarded.py:        assert request.host == f"{url.host}:{url.port}"
tests/components/http/test_forwarded.py:        assert request.host == f"{url.host}:{url.port}"
tests/components/http/test_forwarded.py:        assert request.host == f"{url.host}:{url.port}"
tests/components/http/test_forwarded.py:        assert request.host == f"{url.host}:{url.port}"
tests/components/http/test_forwarded.py:        assert request.host == f"{url.host}:{url.port}"
tests/components/http/test_forwarded.py:        assert request.host == f"{url.host}:{url.port}"
tests/components/http/test_forwarded.py:        assert request.host == "example.com"
tests/components/http/test_forwarded.py:        assert request.host == f"{url.host}:{url.port}"
```

Its also used for `X-Forwarded-Host` which is also a bit up for debate if it should include the port https://github.com/akka/akka-http/issues/2191 details the struggle

**Comment by ffissore** [comment]
Should the `yarl` dependency be pinned to `1.12.1` in https://github.com/aio-libs/aiohttp/blob/master/requirements/runtime-deps.in#L11C1-L11C22 until the issue is solved?

**Comment by Dreamsorcerer** [comment]
I can't imagine we'll make another release without solving the issue, so updating the pin is not going to help.

**Comment by ffissore** [comment]
The problem is that, being `yarl` specified as a range, even those that don't update aiohttp will be affected.
I'm the last one demanding open source devs to "fix this now!", hence my suggestion to pin yarl to buy you time.

**Comment by bdraco** [comment]
I think we’ll do a new release as soon as we can fix this one. I expect I’ll have time this weekend to make that happen.

**Comment by bdraco** [comment]
https://github.com/aio-libs/aiohttp/pull/9318 will fix it for 3.10

The caveat emptor is the fix makes the url call a lot slower so if it matters for you when we ship the fix I'd update yarl once its available as it will shave off most of the performance slowdown in 1.13.1 https://github.com/aio-libs/yarl/pull/1164

**Comment by bdraco** [comment]
I was hoping to release this fix today in 3.10.7 but docker registry is down so the release won't publish

https://www.dockerstatus.com/
